### ISSUE TITLE
Tener un code smell para comparaciones redundantes

### ISSUE THREAD

**Issue by Gustrucco** [mumuki/mulang#230]
Por ejemplo:

`elemento < otro_elemento && elemento != otro_elemento`

**Comment by flbulgarelli** [comment]
¿Cuales serían todas las comparaciones redundantes @Gustrucco? 

Anoto algunas que se me ocurren:

* `elemento < otro_elemento && elemento != otro_elemento`  => sobra el distinto
* `elemento > otro_elemento && elemento != otro_elemento` => sobra el distinto
* `elemento < otro_elemento || elemento != otro_elemento`  => sobra el menor
* `elemento > otro_elemento || elemento != otro_elemento` => sobra el mayor
* `elemento <= otro_elemento || elemento == otro_elemento` => sobra el igual
* `elemento >= otro_elemento || elemento == otro_elemento` => sobra el igual
* `elemento <= otro_elemento && elemento == otro_elemento` => sobra el menor o igual
* `elemento >= otro_elemento && elemento == otro_elemento` => sobra el mayor o igual

**Comment by flbulgarelli** [comment]
Generalización: salvo los casos imposibles  (`elemento >/< otro_elemento && elemento == otro_elemento`) sería: 

```
elemento COMPARACION otro_elemento OP_LOGICA elemento COMPARACION otro_elemento
```

¿No? @Gustrucco @julian-berbel

**Comment by Gustrucco** [comment]
Correcto. En los casos de la conjunción te quedas con el que conjunto más acotado y de la disyunción con el más amplio, ¿correcto?
